openmmtools/forcefactories.py

```diff
@@ -174,6 +174,240 @@ def restrain_atoms(thermodynamic_state, sampler_state, restrained_atoms, sigma=3
+    restricted to water-solute and solute-solute interactions. Water-water
+    interactions are in force group 0, and all other interactions are in force
+    group 1.
+    """
+
+    # create a copy of the original system: only touch this
+    new_system = copy.deepcopy(system)
+
+    # find the default nonbonded force
+    force_index, nb_force = forces.find_forces(new_system, openmm.NonbondedForce, only_one=True)
+    # create copies for each interaction. Only half in solvent/solvent and solute/solute as we double-count.
+    nb_only_solvent_solvent = clone_nonbonded_parameters(nb_force, energy_prefactor='0.5*')
+    nb_solute = clone_nonbonded_parameters(nb_force)
+
+    # NOTE: these need to be python ints -- not np.int64s -- when passing to addInteractionGroup later!
+    solvent_indices = list(map(int, md_topology.select('water')))
+    solute_indices = list(map(int, md_topology.select('not water')))
+
+    #all_indices = list(range(md_topology.n_atoms))
+
+    nb_only_solvent_solvent.addInteractionGroup(set1=solvent_indices, set2=solvent_indices)
+
+    nb_solute.addInteractionGroup(set1=solute_indices, set2=solute_indices)
+    nb_solute.addInteractionGroup(set1=solute_indices, set2=solvent_indices)
+
+    #nb_solute.addInteractionGroup(set1=solute_indices, set2=all_indices)
+
+    # remove original force, add new forces
+    new_system.removeForce(force_index)
+    new_system.addForce(nb_solute)
+    new_system.addForce(nb_only_solvent_solvent)
+
+    # Set solvent-solvent to fg 0, everything else to fg1
+    nb_only_solvent_solvent.setForceGroup(0)
+    nb_solute.setForceGroup(1)
+
+    # handle non-NonbondedForce's
+    for force in new_system.getForces():
+        if 'Nonbonded' not in force.__class__.__name__:
+            force.setForceGroup(1)
+
+    return new_system
+
+
+def split_nb_using_exceptions(system, md_topology):
+    """Construct a new system where force group 0 contains slow, expensive solvent-solvent
+    interactions, and force group 1 contains fast, cheaper solute-solute and solute-solvent
+    interactions.
+
+    TODO: correct this:
+    Force group 0: default nonbonded force with exceptions for solute-{solute,solvent} pairs
+    Force group 1:
+    * CustomNonbonded force with interaction group for solute-{solute,solvent} pairs
+    * non-Nonbonded forces
+
+    TODO: more informative docstring
+    """
+
+    # create a copy of the original system: only touch this
+    new_system = copy.deepcopy(system)
+
+    # find the default nonbonded force
+    force_index, nb_force = forces.find_forces(new_system, openmm.NonbondedForce, only_one=True)
+    # assert('Cutoff' in nb_force.getNonbondedMethod()) # TODO: this, less jankily
+
+    # create copies for each interaction. Only half in solute/solute as we double count.
+    nb_only_solvent_solvent = nb_force
+    nb_only_solvent_solute = clone_nonbonded_parameters(nb_force)
+    nb_only_solute_solute = clone_nonbonded_parameters(nb_force,energy_prefactor='0.5*')
```

Comment:
Why would you multiply the energy by 0.5?

Reply:
Mistakenly assumed double-counting -- will revert